backend/src/controllers/user.js

```diff
@@ -112,5 +112,5 @@ exports.chargeBalance = async (req, res) => {
       console.log("Transaction failed: "+error)
     });
 
-  return res.status(200).json({balance: user.balance});
+  return res.status(200).json({balance: user.balance + req.body.amount});
```

Comment:
Instead of this quick hack you could use {new: true} in the options of findOneAndUpdate to return the new document and leave it as user.balance

Reply:
Ok, thanks for the advice